src/embed.ts

```diff
@@ -94,40 +95,92 @@ async function getVimeoThumbnail(url: URL, sentry: SentryReporter) {
     'https://vimeo.com/api/oembed.json?url=' +
       encodeURIComponent(`https://vimeo.com/${videoId}`)
   )
+  const apiResponseText = await apiResponse.text()
+
+  if (apiResponse.status !== 200) {
+    if (apiResponse.status !== 404) {
+      sentry.setContext(
+        'apiResponse',
+        responseToContext({ response: apiResponse, text: apiResponseText })
+      )
+      sentry.captureMessage(
+        'Request to Vimeo API was not successful',
+        'warning'
+      )
+    }
+    return getPlaceholder()
+  }
 
-  if (apiResponse.status !== 200) return getPlaceholder()
+  let apiResponseJson: unknown = undefined
 
   try {
-    const returnedJson = (await apiResponse.json()) as unknown
+    apiResponseJson = JSON.parse(apiResponseText) as unknown
+  } catch (e) {
+    sentry.setContext(
+      'apiResponse',
+      responseToContext({ response: apiResponse, text: apiResponseText })
+    )
+    sentry.captureMessage('Vimeo API returns malformed JSON', 'warning')
+    return getPlaceholder()
+  }
 
-    if (!VimeoApiResponse.is(returnedJson)) {
-      sentry.setContext('returnedJson', returnedJson)
-      sentry.captureMessage('Vimeo API returns malformed JSON', 'warning')
-      return getPlaceholder()
-    }
+  if (!VimeoApiResponse.is(apiResponseJson)) {
+    sentry.setContext(
+      'apiResponse',
+      responseToContext({ response: apiResponse, json: apiResponseJson })
+    )
+    sentry.captureMessage('Vimeo API returns unsupported JSON', 'warning')
+    return getPlaceholder()
+  }
 
-    const url = returnedJson.thumbnail_url.replace(/_[0-9|x]+/, '')
+  let imgUrl: Url
+  const vimeoThumbnailUrl = apiResponseJson.thumbnail_url.replace(
```

Comment:
aside: why is this necessary, I don't remember?

Reply:
I have no idea either :see_no_evil: Might want to use the returned thumbnail url...

Reply:
well okay, we'll figure it out if there is a problem with this regex at some point ;)